### ISSUE TITLE
Noobie mistakes

### ISSUE THREAD

**Issue by TinCanTech** [koalaman/shellcheck#2365]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): Online
- [ ] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [X] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [ ] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

## not caught: Missing `$` when foo="{bar}"

```sh

#!/bin/sh

foo="{bar}"
echo ${foo}


```

#### Here's what shellcheck currently says:

$ shellcheck myscript
No issues detected!

#### Here's what I wanted or expected to see:

This took me about 30 minutes to spot today.

Shellcheck could catch this ..
